app/controllers/concerns/verify_profile_concern.rb

```diff
@@ -0,0 +1,24 @@
+module VerifyProfileConcern
+  private
+
+  def finish_profile_route
+    decorated_user = current_user.decorate
+    return 'profile' unless decorated_user.pending_profile_requires_verification?
+    return 'verify_profile_phone' if decorated_user.needs_profile_phone_verification?
+    return 'verify_profile' if decorated_user.needs_profile_usps_verification?
+    'profile'
+  end
+
+  def finish_profile_path
```

Comment:
I agree about the naming.

The original method was just `profile_path` and I needed something more nuanced. Maybe it's the `profile_path`. Maybe it's some derivation of `profile_path` depending on the status of your `Profile` record (pending vs active). If it's pending, then the route shunts you to finish verifying your profile, making it active.

So it isn't necessarily `finish_verifying_profile_path`. It's more like, `find_the_next_path_either_to_profile_or_to_finish_verifying_profile`